command-expansion-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -2,46 +2,122 @@
 export class Command<A extends ArgType[] | { [argName: string]: any } = [] | {}> {
   public readonly stem: string;
   public readonly arguments: A;
+  public readonly absoluteTime?: Temporal.Duration;
+  public readonly relativeTime?: Temporal.Duration;
+  public readonly epochTime?: { epochName: string; time: Temporal.Duration };
 
-  private constructor(opts: { stem: string; arguments: A }) {
+  private constructor(opts: {
+    stem: string;
+    arguments: A;
+    absoluteTime?: Temporal.Duration;
+    relativeTime?: Temporal.Duration;
+    epochTime?: { epochName: string; time: Temporal.Duration };
+  }) {
     this.stem = opts.stem;
     this.arguments = opts.arguments;
+    this.absoluteTime = opts.absoluteTime;
+    this.relativeTime = opts.relativeTime;
+    this.epochTime = opts.epochTime;
   }
 
-  public static new<A extends any[] | { [argName: string]: any }>(opts: { stem: string; arguments: A }): Command<A> {
+  public static new<A extends any[] | { [argName: string]: any }>(opts: {
+    stem: string;
+    arguments: A;
+    absoluteTime?: Temporal.Duration;
+    relativeTime?: Temporal.Duration;
+    epochTime?: { epochName: string; time: Temporal.Duration };
+  }): Command<A> {
     return new Command({
       stem: opts.stem,
       arguments: opts.arguments,
+      absoluteTime: opts.absoluteTime,
+      relativeTime: opts.relativeTime,
+      epochTime: opts.epochTime,
     });
   }
 
   public toSeqJson() {
-    return {
+    let seqJson: any = {
       id: 'command',
       metadata: {},
+
       steps: [
         {
           stem: this.stem,
-          time: { type: 'COMPLETE' },
+          time: {
+            tag: this.absoluteTime
+              ? this.absoluteTime.toString()
+              : this.relativeTime
+              ? this.relativeTime.toString()
+              : this.epochTime
+              ? this.epochTime.time.toString()
+              : '',
+            type: this.absoluteTime
+              ? 'ABSOLUTE'
+              : this.relativeTime
+              ? 'COMMAND_RELATIVE'
+              : this.epochTime
+              ? 'EPOCH_RELATIVE'
+              : 'COMMAND_COMPLETE',
+          },
           type: 'command',
           metadata: {},
           args: typeof this.arguments == 'object' ? Object.values(this.arguments) : this.arguments,
         },
       ],
     };
+    // include epoch field if we have an epoch time
+    if (this.epochTime) {
+      seqJson['epoch'] = this.epochTime.epochName;
```

Comment:
we can put this in above with a ternary like so
```
{
       id: 'command',
       metadata: {},
       ...(this.epochTime !== null ? { epoch: this.epochTime.epochName } : {})
       steps: [
         {
```

I think it's a little more clear that we have a conditional attribute that way

Reply:
Or we can put it in and strip undefined attributes prior to returning.